drivers/mipi_dsi/dsi_renesas_ra.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra_mipi_dsi
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/renesas_ra_cgc.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mipi_dsi.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/irq.h>
+#include <zephyr/logging/log.h>
+#include "r_mipi_dsi.h"
+#include "r_mipi_phy.h"
+
+LOG_MODULE_REGISTER(dsi_renesas_ra, CONFIG_MIPI_DSI_LOG_LEVEL);
+
+/* MIPI PHY Macros */
+#define MIPI_PHY_CLKSTPT (1183)
+#define MIPI_PHY_CLKBFHT (11)
+#define MIPI_PHY_CLKKPT  (26)
+#define MIPI_PHY_GOLPBKT (40)
+
+#define MIPI_PHY_TINIT     (71999)
+#define MIPI_PHY_TCLKPREP  (8)
+#define MIPI_PHY_THSPREP   (5)
+#define MIPI_PHY_TCLKTRAIL (7)
+#define MIPI_PHY_TCLKPOST  (19)
+#define MIPI_PHY_TCLKPRE   (1)
+#define MIPI_PHY_TCLKZERO  (27)
+#define MIPI_PHY_THSEXIT   (11)
+#define MIPI_PHY_THSTRAIL  (8)
+#define MIPI_PHY_THSZERO   (19)
+#define MIPI_PHY_TLPEXIT   (7)
+#define LP_DIVISOR         (4)
+#define PLL_MUL_SETTING    (49)
+#define VIDEO_MODE_DELAY   (186)
+#define ULPS_WAKEUP_PERIOD (97)
+struct mipi_dsi_renesas_ra_config {
+	const struct device *clock_dev;
+	struct clock_control_ra_subsys_cfg clock_dsi_subsys;
+	void (*irq_configure)(void);
+};
+
+struct mipi_dsi_renesas_ra_data {
+	mipi_dsi_instance_ctrl_t mipi_dsi_ctrl;
+	mipi_dsi_cfg_t mipi_dsi_cfg;
+};
+
+volatile bool g_message_sent;
```

Comment:
Don't use global variables like these. Store these within the device data structure to properly support multi instance drivers